### PR TITLE
[CARBONDATA-4192] UT cases correction for validating the exception message correctly

### PR DESCRIPTION
**Pull request by maheshrajus**
### Why is this PR needed?
 Currently, when we check the exception message like below, it is not asserting/failing/catching if the message content is different.
```
intercept[UnsupportedOperationException](
sql("update test set(a)=(4) where id=1").collect()).getMessage.contains("abc")
```

 ### What changes were proposed in this PR?
1) Added assert condition like below for validating the exception message correctly
```
assert(intercept[UnsupportedOperationException](
sql("update test set(a)=(4) where id=1").collect()).getMessage.contains("abc"))
```
2) Added assert condition to check exception message for some test cases which are not checking exception message
3) Fixed add segment doc heading related issues
 ### Does this PR introduce any user interface change?
 - No
 ### Is any new testcase added?
 - No

### CODE REVIEW

**Comment by CarbonDataQA2** [comment]
Build Success with Spark 2.4.5, Please check CI http://121.244.95.60:12602/job/ApacheCarbon_PR_Builder_2.4.5/3681/

**Comment by CarbonDataQA2** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12602/job/ApacheCarbonPRBuilder2.3/5426/

File: docs/addsegment-guide.md [hunk 1]
```diff
@@ -37,12 +37,12 @@ In the above command user can add the existing data to the carbon table as a new
 During add segment, it will infer the schema from data and validates the schema against the carbon table. 
 If the schema doesn’t match it throws an exception.
 
-###Changes to tablestatus file
+### Changes to tablestatus file
```
Comment (Indhumathi27) [inline docs/addsegment-guide.md:40]: Please change line 441 ‘format’  to `format`

File: docs/addsegment-guide.md [hunk 2]
```diff
@@ -51,7 +51,7 @@ To support multiple formats per segment basis we can create multiple RDD using t
 Note: This integration will be clean as we use the sparks optimized reading, pruning and it
```
Comment (Indhumathi27) [inline docs/addsegment-guide.md:51]:
```suggestion
**Note**: This integration will be clean as we use the sparks optimized reading, pruning and it
```

File: docs/addsegment-guide.md [hunk 3]
```diff
@@ -17,14 +17,14 @@
...
 In the industry, many users already adopted to data with different formats like ORC, Parquet, JSON, CSV etc.,  
 If users want to migrate to Carbondata for better performance or for better features then there is no direct way. 
 All the existing data needs to be converted to Carbondata to migrate.  
 This solution works out if the existing data is less, what if the existing data is more?   
 Heterogeneous format segments aims to solve this problem by avoiding data conversion.
 
-###Add segment with path and format
+### Add segment with path and format
 Users can add the existing data as a segment to the carbon table provided the schema of the data
  and the carbon table should be the same.
```
Comment (Indhumathi27) [inline docs/addsegment-guide.md:30]: Looks like syntax is wrong. Should be
Alter table table_name add segment options (‘path’= 'hdfs://usr/oldtable,'format'='parquet')

**Comment by Indhumathi27** [comment]
@maheshrajus Still, one testclass is missed. Check DeleteCarbonTableTestCase

File: docs/addsegment-guide.md (hunk 1 above)
Comment (maheshrajus) [reply docs/addsegment-guide.md:40]: done

File: docs/addsegment-guide.md (hunk 2 above)
Comment (maheshrajus) [reply docs/addsegment-guide.md:51]: done

File: docs/addsegment-guide.md (hunk 3 above)
Comment (maheshrajus) [reply docs/addsegment-guide.md:30]: done

**Comment by CarbonDataQA2** [comment]
Build Failed  with Spark 2.4.5, Please check CI http://121.244.95.60:12602/job/ApacheCarbon_PR_Builder_2.4.5/3737/

**Comment by CarbonDataQA2** [comment]
Build Failed  with Spark 2.3.4, Please check CI http://121.244.95.60:12602/job/ApacheCarbonPRBuilder2.3/5481/

**Comment by maheshrajus** [comment]
retest this please

**Comment by CarbonDataQA2** [comment]
Build Failed  with Spark 2.3.4, Please check CI http://121.244.95.60:12602/job/ApacheCarbonPRBuilder2.3/5486/

**Comment by maheshrajus** [comment]
retest this please

**Comment by CarbonDataQA2** [comment]
Build Success with Spark 2.4.5, Please check CI http://121.244.95.60:12602/job/ApacheCarbon_PR_Builder_2.4.5/3742/

File: docs/addsegment-guide.md [hunk 4]
```diff
@@ -17,41 +17,41 @@
...
 In the above command user can add the existing data to the carbon table as a new segment and also
  can provide the data format.
 
 During add segment, it will infer the schema from data and validates the schema against the carbon table. 
 If the schema doesn’t match it throws an exception.
 
-###Changes to tablestatus file
-Carbon adds the new segment by adding segment information to tablestatus file. In order to add the path and format information to tablestatus, we are going to add `segmentPath`  and ‘format’  to the tablestatus file. 
+### Changes to tablestatus file
+Carbon adds the new segment by adding segment information to tablestatus file. In order to add the path and format information to tablestatus, we are going to add `segmentPath`  and format  to the tablestatus file.
```
Comment (Indhumathi27) [inline docs/addsegment-guide.md:41]:
```suggestion
Carbon adds the new segment by adding segment information to tablestatus file. In order to add the path and format information to tablestatus, we are going to add `segmentPath`  and `format`  to the tablestatus file. 
```

**Comment by CarbonDataQA2** [comment]
Build Success with Spark 2.4.5, Please check CI http://121.244.95.60:12602/job/ApacheCarbon_PR_Builder_2.4.5/3746/

**Comment by CarbonDataQA2** [comment]
Build Failed  with Spark 2.3.4, Please check CI http://121.244.95.60:12602/job/ApacheCarbonPRBuilder2.3/5490/

**Comment by Indhumathi27** [comment]
LGTM

**Comment by Indhumathi27** [comment]
retest this please

**Comment by CarbonDataQA2** [comment]
Build Failed  with Spark 2.4.5, Please check CI http://121.244.95.60:12602/job/ApacheCarbon_PR_Builder_2.4.5/3749/

**Comment by CarbonDataQA2** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12602/job/ApacheCarbonPRBuilder2.3/5492/

**Comment by Indhumathi27** [comment]
retest this please

**Comment by CarbonDataQA2** [comment]
Build Success with Spark 2.4.5, Please check CI http://121.244.95.60:12602/job/ApacheCarbon_PR_Builder_2.4.5/3751/

**Comment by CarbonDataQA2** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12602/job/ApacheCarbonPRBuilder2.3/5494/

**Pull request closed without merging**